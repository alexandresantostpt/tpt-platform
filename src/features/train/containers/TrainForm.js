import { connect } from 'react-redux'

import { getObjSelector as travelGetObjSelector, getClientsSelector } from '@features/travelScript/selectors'

import { getObjSelector } from '../selectors'
import { requestGetSingleAction, requestSaveDataAction, requestSearchAction } from '../actions'

import TrainForm from '../pages/TrainForm'

const mapStateToProps = state => ({
    clients: getClientsSelector(state),
    train: getObjSelector(state),
    travelScript: travelGetObjSelector(state)
})

const mapDispatchToProps = {
    requestGetSingleAction,
    requestSaveDataAction,
    requestSearchAction
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(TrainForm)
