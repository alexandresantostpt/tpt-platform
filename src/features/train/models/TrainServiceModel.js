import { config } from '@config'

import moment from 'moment'

import { normalizeTimezone } from '@utils/date'

import TrainLibraryModel from './TrainLibraryModel'

class TrainServiceModel {
    constructor(
        {
            __v,
            _id,
            cityFrom,
            cityDestiny,
            companyName,
            createdAt,
            dateFrom,
            destiny,
            from,
            hourEstimatedArrival,
            hourFrom,
            library,
            passengers,
            passengersValues,
            passengerName,
            scriptDate,
            stationDestiny,
            stationFrom,
            updatedAt
        },
        fromForm = false
    ) {
        this.__v = __v
        this._id = _id
        this._cityFrom = cityFrom
        this._cityDestiny = cityDestiny
        this._companyName = companyName
        this._createdAt = moment(createdAt).utc(false)
        this._dateFrom = moment(dateFrom).utc(false)
        this._destiny = destiny
        this._from = from
        this._hourEstimatedArrival = fromForm ? normalizeTimezone(hourEstimatedArrival) : moment(hourEstimatedArrival).utc(false)
        this._hourFrom = fromForm ? normalizeTimezone(hourFrom) : moment(from && from.hour).utc(false)
        this._library = library ? new TrainLibraryModel(library) : null
        this._passengers = passengers
        this._passengersValues = passengersValues
        this._passengerName = passengerName
        this._scriptDate = scriptDate
        this._stationDestiny = stationDestiny
        this._stationFrom = stationFrom
        this._updatedAt = moment(updatedAt).utc(false)
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get cityFrom() {
        return this._cityFrom
    }

    get cityDestiny() {
        return this._cityDestiny
    }

    get companyName() {
        return this._companyName
    }

    get createdAt() {
        return this._createdAt
    }

    get dateFrom() {
        return this._dateFrom
    }

    get destiny() {
        return this._destiny
    }

    get from() {
        return this._from
    }

    get hourFrom() {
        return this._hourFrom
    }

    get hourEstimatedArrival() {
        return this._hourEstimatedArrival
    }

    get library() {
        return this._library
    }

    get passengers() {
        return this._passengers
    }

    get passengersValues() {
        return this._passengersValues
    }

    get passengerName() {
        return this._passengerName
    }

    get scriptDate() {
        return this._scriptDate
    }

    get stationDestiny() {
        return this._stationDestiny
    }

    get stationFrom() {
        return this._stationFrom
    }

    get updatedAt() {
        return this._updatedAt
    }

    getImage() {
        return this.library && this.library.image && `${config.api.url}/train/download/${this.library.image}`
    }

    /* eslint-disable-next-line complexity */
    toDTO() {
        return {
            __v: this.v,
            _id: this.id,
            companyName: this.companyName || null,
            destiny: {
                city: this.cityDestiny || null,
                state: this.stationDestiny || null
            },
            from: {
                city: this.cityFrom || null,
                date: this.dateFrom || null,
                hour: this.hourFrom || null,
                state: this.stationFrom || null
            },
            hourEstimatedArrival: this.hourEstimatedArrival || null,
            library: this.library ? this.library.toDTO() : null,
            passengers: [...this.passengers.map(item => this.passengersValues.find(passenger => passenger.client === item))],
            scriptDate: this.scriptDate
        }
    }

    toForm() {
        return {
            __v: this.v,
            _id: this.id,
            cityDestiny: this.destiny.city,
            cityFrom: this.from.city,
            companyName: this.companyName,
            dateFrom: this.from.date,
            hourEstimatedArrival: this.hourEstimatedArrival,
            hourFrom: this.hourFrom,
            library: this.library ? this.library.toForm() : null,
            passengers: [...this.passengers.map(item => item.client)],
            passengersValues: this.passengers,
            platform: this.platform,
            scriptDate: this.scriptDate,
            seat: this.seat,
            stationDestiny: this.destiny.state,
            stationFrom: this.from.state,
            wagon: this.wagon
        }
    }
}

export default TrainServiceModel
