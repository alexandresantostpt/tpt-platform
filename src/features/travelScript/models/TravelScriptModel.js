import moment from 'moment'

import Travel from '../../travel/TravelModel'
import TravelDateModel from './TravelDateModel'
import { isNotEmpty } from '@/utils/functions'

class TravelScriptModel {
    constructor({ __v, _id, dates, deleted, edited, travel, createdAt, updatedAt }) {
        this.__v = __v
        this._id = _id
        this._dates = dates && [...dates].map(date => new TravelDateModel(date))
        this._deleted = deleted
        this._edited = edited
        this._travel = new Travel(travel)
        this._createdAt = moment(createdAt).utc(false)
        this._updatedAt = moment(updatedAt).utc(false)
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get dates() {
        return this._dates
    }

    get deleted() {
        return this._deleted
    }

    get edited() {
        return this._edited
    }

    get travel() {
        return this._travel
    }

    get createdAt() {
        return this._createdAt
    }

    get updatedAt() {
        return this._updatedAt
    }

    hasServices() {
        return this.dates.some(date => date.services.length)
    }

    getRangeDates() {
        if (isNotEmpty(this._travel)) {
            return { endDate: this._travel.travelEndDate, startDate: this._travel.travelStartDate }
        }

        return {}
    }

    getClientsId() {
        return [...this.travel.clients].map(({ _id }) => _id)
    }

    toForm() {
        return {
            dates: this.dates,
            edited: this.edited,
            id: this.id,
            travel: this.travel,
            v: this.v
        }
    }
}

export default TravelScriptModel
