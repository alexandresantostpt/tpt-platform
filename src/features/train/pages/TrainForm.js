import { events } from '@constants/events'

import PropTypes from 'prop-types'
import PubSub from 'pubsub-js'
import React, { useEffect, useRef, useState } from 'react'

import { colors } from '@helpers/colors'
import { getQueryParams } from '@utils/browser'
import { i18n } from '@i18n'
import { services } from '@/constants/services'
import { withRouter } from 'react-router-dom'

import Button from '@components/button/Button'
import Cancel from '@components/cancel/Cancel'
import FormBuilder from '@components/form/FormBuilder'
import Panel from '@components/panel/Panel'

import { trainFormFields } from '../fields'

import FormButtonContainer from '@components/form/container/button/FormButtonContainer'
import WrapperPadding from '@components/wrapper/WrapperPadding'

import TrainClientForm from '../components/trainClientForm/TrainClientForm'
import TrainServiceModel from '../models/TrainServiceModel'

const TrainForm = props => {
    const { train, match, travelScript } = props
    const {
        params: { id, idTrain }
    } = match
    const { dateId, scriptDate, scriptId } = getQueryParams()

    const form = useRef()
    const previousPath = `/travel/${id}/script`

    const [image, updateImage] = useState()
    const [serviceDisabled, updateServiceDisabled] = useState(false)

    useEffect(() => {
        const { requestGetSingleAction } = props
        if (idTrain) {
            requestGetSingleAction({ id: idTrain })
            updateServiceDisabled(true)
        } else {
            requestGetSingleAction({ id: null })
            updateServiceDisabled(false)
        }
    }, [idTrain])

    const callSaveData = values => {
        const { requestSaveDataAction } = props
        requestSaveDataAction({
            data: {
                dateId,
                scriptId,
                service: new TrainServiceModel({ ...values, scriptDate }, true).toDTO(),
                type: services.TRAIN
            },
            image,
            redirectTo: previousPath
        })
        reset()
    }

    const onSaveClick = () => form.current.form.submit()

    const reset = () => {
        PubSub.publish(events.FORM_RESET)
        updateImage(null)
    }

    return (
        <>
            <WrapperPadding>
                <Panel backgroundImageUrl={train && train.getImage()} icon="train" uploadImage={updateImage} />
            </WrapperPadding>
            <FormBuilder
                clients={props.clients}
                disabled={serviceDisabled}
                fieldChildren={<TrainClientForm clients={props.clients} form={form} />}
                fields={trainFormFields}
                initialValues={train ? train.toForm() : { dateFrom: scriptDate }}
                onSubmit={callSaveData}
                rangeDates={travelScript && travelScript.getRangeDates()}
                ref={process.env.__TEST__ ? null : form}
            />
            <FormButtonContainer>
                <Button
                    color={colors.purple}
                    onClick={() => (serviceDisabled ? updateServiceDisabled(false) : onSaveClick())}
                    text={i18n.t(serviceDisabled ? 'buttons.editService' : 'buttons.saveService')}
                />
                <Cancel link={previousPath} />
            </FormButtonContainer>
        </>
    )
}

TrainForm.defaultProps = {
    clients: [],
    train: {},
    travelScript: {}
}

TrainForm.propTypes = {
    clients: PropTypes.array,
    match: PropTypes.object.isRequired,
    requestGetSingleAction: PropTypes.func.isRequired,
    requestSaveDataAction: PropTypes.func.isRequired,
    train: PropTypes.objectOf(TrainServiceModel),
    travelScript: PropTypes.object
}

export default withRouter(TrainForm)
