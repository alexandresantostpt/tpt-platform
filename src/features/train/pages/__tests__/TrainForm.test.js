import React from 'react'
import renderer from 'react-test-renderer'

import TrainForm from '../TrainForm'

import TrainServiceModel from '../../models/TrainServiceModel'
import TravelScriptModel from '@/features/travelScript/models/TravelScriptModel'

describe('Tests for TrainForm component', () => {
    const clients = [
        {
            __v: 0,
            _id: '5c9b6c386ffd761521bd80df',
            active: false,
            blocked: false,
            cpf: '123.123.123-12',
            createdAt: '2019-03-27T12:27:36.838Z',
            email: 'teste@teste.com',
            name: 'Viajante teste',
            password: '1b621fe4e8bd896fa918088696a393ab2330ee73373efe770833b85598756a28',
            updatedAt: '2019-03-27T12:27:36.838Z'
        }
    ]
    const train = new TrainServiceModel({
        __v: 0,
        _id: '5c9b70786ffd767caabd80e5',
        companyName: 'Nome da companhia',
        createdAt: '2019-03-27T12:45:44.438Z',
        destiny: { city: 'teste', state: 'teste' },
        from: {
            city: 'Cidade de origem',
            date: '2019-03-27T00:00:00.000Z',
            hour: '2019-03-27T13:00:00.489Z',
            state: 'Estação de origem'
        },
        hourEstimatedArrival: '2019-03-27T11:00:00.692Z',
        passengers: [
            { _id: '5c9b70786ffd763279bd80e6', client: '5c9b6c386ffd761521bd80df', platform: 'A1', seat: '11B', wagon: '22' }
        ],
        script: '5c9b6c386ffd76125cbd80e1',
        scriptDate: '2019-03-27T00:00:00.000Z',
        type: 'TRAIN',
        updatedAt: '2019-03-27T12:45:44.438Z'
    })
    const travelScript = new TravelScriptModel({ travel: { __v: 0 } })

    it('Should render withou crash', () => {
        const match = {
            params: {
                id: '5c52ebcd5fc7c949223ab705'
            }
        }
        const tree = renderer
            .create(
                <TrainForm
                    clients={clients}
                    match={match}
                    requestDelAction={jest.fn()}
                    requestGetListAction={jest.fn()}
                    requestGetSingleAction={jest.fn()}
                    requestSearchAction={jest.fn()}
                    train={train}
                    travelScript={travelScript}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
