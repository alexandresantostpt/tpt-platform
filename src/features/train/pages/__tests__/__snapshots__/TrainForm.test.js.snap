// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for TrainForm component Should render withou crash 1`] = `
Array [
  <div
    className="WrapperPadding-ovjbyx-0 lcnoja"
  >
    <div
      className="Panel__StyledPanel-sc-1qzkpb6-0 cGUlCn"
    >
      <span
        className="Panel__StyledIcon-sc-1qzkpb6-2 iavvSk"
      >
        <span
          className="icon icon-train Icon__StyledIcon-crr1cg-0 dfhcZR"
          disabled={false}
          fontSize="3rem"
          onClick={null}
        />
      </span>
      <div
        className="UploadButton__StyledContainer-sc-1p8zoc-0 fpZFIG"
      >
        <label
          className="UploadButton__StyledUploadButton-sc-1p8zoc-3 ivhrYW"
          htmlFor="inputFile"
        >
          +
        </label>
        <input
          accept="image/png, image/jpeg"
          className="UploadButton__InputFile-sc-1p8zoc-4 eLaooA"
          id="inputFile"
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
    <ul
      className="PanelActions__StyledActions-sc-8xk8c8-0 lllkGw"
    />
  </div>,
  <form
    className="FormBuilder__StyledForm-c1htou-0 PZxQa"
    onSubmit={[Function]}
  >
    <div
      className="form__container"
    >
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Nome da companhia
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.name"
          disabled={false}
          id="companyName"
          maxLength={120}
          name="library.name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Cidade de origem
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-cityFrom"
          disabled={false}
          id="cityFrom"
          maxLength={120}
          name="cityFrom"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Cidade de origem"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Estação de origem
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-stationFrom"
          disabled={false}
          id="stationFrom"
          maxLength={60}
          name="stationFrom"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Estação de origem"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Data de saída
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormDate__StyledFormDate-sc-1cgz0rk-0 ebKpCI"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-"
                    aria-label="Selecione uma data..."
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                    disabled={false}
                    id=""
                    name=""
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Selecione uma data..."
                    readOnly={true}
                    required={false}
                    type="text"
                    value="27 Mar 2019"
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon icon-calendar Icon__StyledIcon-crr1cg-0 cmjVJQ"
                    disabled={false}
                    fontSize="1.75rem"
                    onClick={null}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Horário de saída
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="10:00"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Cidade de destino
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-cityDestiny"
          disabled={false}
          id="cityDestiny"
          maxLength={120}
          name="cityDestiny"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="teste"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Estação de destino
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-stationDestiny"
          disabled={false}
          id="stationDestiny"
          maxLength={60}
          name="stationDestiny"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="teste"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Horário estimado de chegada
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="11:00"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Passageiros
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormClients__StyledOption-jb3pw2-1 gugIhy"
        >
          <input
            checked={true}
            disabled={false}
            id="5c9b6c386ffd761521bd80df"
            name="5c9b6c386ffd761521bd80df"
            onChange={[Function]}
            type="checkbox"
            value="5c9b6c386ffd761521bd80df"
          />
          <span
            className="FormClients__StyledLabel-jb3pw2-0 kKzFxc"
          >
            Viajante teste
          </span>
        </div>
        <div
          className="TrainClientForm__StyledTrainClientForm-sc-1i1l77e-0 iNQVZV"
        >
          <span>
            <div
              className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
            >
              <label
                className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
              >
                Nome do passageiro
                 1
              </label>
              <input
                className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                data-cy="input-passengersValues[0].passengerName"
                disabled={false}
                id="passengersValues[0].passengerName"
                maxLength={60}
                name="passengersValues[0].passengerName"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder=""
                readOnly={false}
                required={false}
                type="text"
                value="Viajante teste"
              />
            </div>
            <div
              className="Collapse__StyledCollapseItem-sc-16jlua7-0 hveKjh"
            >
              <div
                className="CollapseItems__StyledCollapseItems-tyowld-0 fnYIc"
              >
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    Plataforma
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-passengersValues[0].platform"
                    disabled={false}
                    id="passengersValues[0].platform"
                    maxLength={30}
                    name="passengersValues[0].platform"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="text"
                    value="A1"
                  />
                </div>
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    Vagão
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-passengersValues[0].wagon"
                    disabled={false}
                    id="passengersValues[0].wagon"
                    maxLength={30}
                    name="passengersValues[0].wagon"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="text"
                    value="22"
                  />
                </div>
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    Assento
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-passengersValues[0].seat"
                    disabled={false}
                    id="passengersValues[0].seat"
                    maxLength={30}
                    name="passengersValues[0].seat"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="text"
                    value="11B"
                  />
                </div>
              </div>
              <span
                className="icon icon-arrow-up Icon__StyledIcon-crr1cg-0 cMlgiC"
                disabled={false}
                fontSize="1.75rem"
                onClick={[Function]}
              />
            </div>
          </span>
        </div>
      </div>
    </div>
  </form>,
  <div
    className="FormButtonContainer-sc-1cqqivn-0 ADyLt"
  >
    <button
      className="Button__StyledButton-sc-1fc3klx-0 bEGBgE"
      disabled={false}
      fill={false}
      onClick={[Function]}
    >
      Salvar serviço
    </button>
    <a
      className="Cancel__StyledCancel-sc-1prgtit-0 WzuoR"
      to="/travel/5c52ebcd5fc7c949223ab705/script"
    >
      Cancelar
    </a>
  </div>,
]
`;
